Adding color file

skanderboudawara opened the pull request:
Answering issue #11

Commit:
Adding color file

Grimmys commented:
Hi, thanks for your first contribution!

While the core principle of your PR seems good to me, I have a few concerns I would like to discuss with you though:

- In `src/colors.py`, after the definition of each color, you are putting a comment to say where it is used. I'm not sure to see in which way it could be useful, and I'm afraid it would be a pain to maintain in the future.
Imagine one day someone decides to use let say the `GREEN` constant in `mini_game_commands.py` because they want to add a new mini-game with a green embed, but they forget to add the related usage command in the `colors.py` file. This mistake would not even be easy to spot at the review, and at the end, in maybe a few months, the comments in the file would not be accurate at all anymore.
Avoiding unnecessary comments, that could be a burden to maintain up-to-date, is a good practice to follow.
- In `main.py`, I believe we could find a more descriptive name for the import of the colors module. I don't really like single-character identifiers. In fact, maybe we could even get rid of the alias and simply import the colors directly:
```python
from src.colors import GREEN, BLUE
[...]
       embed = Embed(title=card.name, description=card_id, color=GREEN)
```
This way, only by reading the import line at the top, it is possible to know exactly which colors are used in the file: green and blue.

Anyway, don't take my comments too seriously, I'm opened to hear your opinion as well if you disagree.

Commit:
adjusted with main user Feedback

skanderboudawara commented:
I agree with both of your comments, and I added the changes.

- I added the comment at first to keep history where the colors were used, but you are right; It will become a mess later on
- For the second point, yes, it's better to import Color by Color for the time being. But imagine tomorrow, if you have to deal with ten colors or more, will you stack them in a trailing comma? I honestly prefer to deal with a letter when it comes to constants (but it's just a matter of preference) -> Another way around this is to deal with dicts() like so
```
color_dict = {
'GREEN': 0x00ff00,
'BLUE': 0x0000FF,
'MAGENTA' : 0xa020f0,
'GRAY': 0x808080,
'YELLOW': 0xffff00 ,
}
```


The PR should be up to date with your new feedback 😁

Grimmys commented:
Awesome, LGTM!

> * For the second point, yes, it's better to import Color by Color for the time being. But imagine tomorrow, if you have to deal with ten colors or more, will you stack them in a trailing comma? I honestly prefer to deal with a letter when it comes to constants (but it's just a matter of preference) -> Another way around this is to deal with dicts() like so

Fair point, I forgot to consider that `colors.py` could be holding way more values in the future due to its nature.
But still, this situation is hypothetical, since good practices also imply to have files with limited scope, meaning that every file should stay quite small (or medium) size, and so few colors should be imported. If a lot have to be used because let's say we want to display something multicolored, then maybe we could just use wildcard import?
Anyway, in this case I would also be totally fine with import using an identifier, but it is just that I prefer to have something meaningful, so for example `Colors`, instead of `C`.
